src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -160,7 +168,7 @@ public void testFromXContent() throws IOException {
     public void testToQuery() throws IOException {
         testQuery = createTestQueryBuilder();
         QueryParseContext context = createContext();
-        context.setMapUnmappedFieldAsString(true);
+        context.setAllowUnmappedFields(true);
```

Comment:
> otherwise the String mapper wins over the date mapper we introduced here

that seems to me like the wrong reason to have it around :) 

Which codepaths are we testing now? unmapped fields only, mapped fields only or both?